### DISCUSSION TITLE
[Suggestion] Add menu to enable and disable every item individually

### DISCUSSION THREAD

**Discussion by 51a2ff** [HotCakeX/Harden-Windows-Security#372] [General]
Hello, @HotCakeX. 
I want to thank you for the Harden Windows Security module. It is amazing and the best in what it proposes. Together with all the documentation.
I want to suggest the capability to enable and disable the items individually.
Why?
1. I like most of the items. But some, I don't. In that case it would be perfect to be able to select individually the ones I don't want enabled as to not have to disable them manually every time I make a new install or every time the tool adds more items.
2. I have encountered some programs that break because of some items. If I have a good idea of what is happening, I can maybe think of the exact item that is breaking the program and figure out how to disable it manually. But for the case described below, I was unable to find the item that breaks the program, because I have no idea of what is breaking it and it would be unfeasible to disable each item and test every time.
3. Manually disabling individual items can be hard because of the lack of information on how to disable them individually.

The latest and most serious problem I had was with the Battle.net desktop app (today).
I was not able to log in it. It could authenticate my login information, ask and confirm my 2FA. But after that, this error always appeared:
![A](https://github.com/user-attachments/assets/f9e196f0-44c7-4c7b-b113-c4bbf7c42ed5)

Nothing appeared in the Protection history of Windows Security. 
I use the latest version of Windows.
I don't have any third-party security apps.
The problem did not happen in Windows Sandbox, using the same computer.
I painfully opted to use Unprotect, first only removing Process Mitigations, then, removing App Control Policies.
Those did not work, so I used Remove All Protections.
It worked. So, some item or items break the app.

Thank you very much for you attention.

**Comment by HotCakeX** [comment]
Hi, thank you very much,

The issue with the battle.net client is mentioned on the readme on the TLS category, it also links to a discussion where i talked about it with someone else. Removing all protections wasn't really necessary but it's okay, you can apply them again any time.
https://github.com/HotCakeX/Harden-Windows-Security?tab=readme-ov-file#tls-security

<img width="895" alt="image" src="https://github.com/user-attachments/assets/5426d4c2-9da3-4bca-bc0f-ab6894baac4a">

<br>

<br>

So before thinking about implementing a separate option for every single 250+ policy that is applied so far (which is a lot of work), let's talk about other problems you are facing so i can have the full picture.

Because for the Battle.net client, i can add a check to the TLS category in the next version to detect Battle.net client on the system automatically and then keep the TLS cipher suite that it needs for connectivity, so user won't have to do anything extra. If you let me know about other problem(s) i can think about better solutions for them as well.

Thanks

**Reply by HotCakeX** [reply]
By the way, in the future it is a possibility to have separate option for every single policy and security measure, but since there are lots of other things currently in development i need to prioritize.

**Comment by 51a2ff** [comment]
Hi.

Thank you for the quick response. 
I feel stupid now, it is in the main page and the issues page, I should have looked better.
Sure, I will compile the other stuff that I had problems with.
I didn't know it would be a lot of work. But I'm happy that you consider it a possibility for the future, it would greatly improve the tool.
Keep up the great work, you will be recognized for it!

Thanks.

**Reply by HotCakeX** [reply]
Ah it's fine, there are lots of stuff to read on readme page so don't worry about it.

Of course, feedback is always welcome

Thank you :)

**Comment by HotCakeX** [comment]
Just released the version 0.6.8 that automatically detects BattleNet client on the system and uses the appropriate cipher suite for the TLS category that allows the client to connect to the servers. All you need to do is to run the same command again that you were using, which automatically updates to the latest version, and then run the TLS category.

**Or** you can just use this command for even more automation :)

```powershell
Protect-WindowsSecurity -Categories TLSSecurity
```

<br>

If you encounter any other problems please open separate issues for each of them so i can address them one by one asap. Thanks!

Here is the release notes for the version 0.6.8: https://github.com/HotCakeX/Harden-Windows-Security/releases/tag/Hardening-Module-v.0.6.8
